src/libasr/pass/array_op.cpp

```diff
@@ -1103,7 +1103,11 @@ class ArrayOpVisitor : public ASR::CallReplacerOnExpressionsVisitor<ArrayOpVisit
                                     }
                                     if ( is_arg ) {
                                         ASR::symbol_t* subrout = subrout_call->m_name;
-                                        if ( subrout->type == ASR::symbolType::Function ) {
+                                        if ( ASR::is_a<ASR::ExternalSymbol_t>(*subrout) ) {
+                                            ASR::ExternalSymbol_t* subrout_ext = ASR::down_cast<ASR::ExternalSymbol_t>(subrout);
+                                            subrout = subrout_ext->m_external;
+                                        }
+                                        if ( ASR::is_a<ASR::Function_t>(*subrout) ) {
```

Comment:
This should be harmless but can be condensed into,

```suggestion
                                        subrout = ASRUtils::symbol_get_past_external(subrout);
                                        if ( ASR::is_a<ASR::Function_t>(*subrout) ) {
```